client/api_object.go

```diff
@@ -159,14 +161,20 @@ func (c *client) CreateObject(ctx context.Context, bucketName, objectName string
 		return "", err
 	}
 
+	ctxTimeout, cancel := context.WithTimeout(ctx, time.Second*30)
```

Comment:
this will override user's setting timeout if there is? And better make it a constant

Reply:
no override, ctx is the parent context , If the parent's deadline is already earlier than the ctxTimeout, it is semantically equivalent to parent.

Reply:
updated to constant

Reply:
Suggest to use `Deadline` function to check whether the timeout already be setted.

```
ctx, cancel := context.WithTimeout(context.Background(), time.Second * 5)
defer cancel()

deadline, ok := ctx.Deadline()
if !ok {
    fmt.Println("No deadline set.")
} else if deadline.Before(time.Now()) {
    fmt.Println("Timeout occurred.")
} else {
    fmt.Println("Not timed out yet.")
}

```

Reply:
The code here should better be simple. The API has enabled a similar function inside the WithTimeout function, The code here should be simple 
<img width="782" alt="image" src="https://github.com/bnb-chain/greenfield-go-sdk/assets/19421226/2bb50994-e7c7-4e0c-8575-2a4b166e6a81">